doc/sync-label.md

```diff
@@ -0,0 +1,224 @@
+
+I would love to sync my personal bug repo with these external trackers but
+there are a couple of problems:
+
+1. Our JIRA at work has millions of bugs. I obviously don't interact with
+   all of them. I only want to sync bugs that are "relevant" to me... where
+   "relevant" can presumably be defined by matching some criteria on metadata
+   in JIRA.
+2. Our gitlab at work likewise has thousands of open pull requests. I'm not sure
+   git-bug is necessarily the best tool for doing code-review right now, but
+   at the very least it might be nice to trach which code-reviews that I'm
+   involved in are still open and require attention.
+3. Some of the bugs in my repo really shouldn't be pushed to a public location
+   because they might contain "sensitive" information about my employer.
+4. I don't want bugs for irrelevant projects to pollute the issue list on
+   my github mirrors.
+
+## Proposed Solution
+
+My proposed solution is to restict synchronization to bugs that match certain
+filters, and to modify bugs as they are imported/exported according to rules
+in the git config. Initially, the filter will support matching labels, and
+modifiers will support adding or removing labels.
+
+NOTE: I may also take a stab at implementing a JIRA bridge.
+
+### Case Study
+
+Consider the project `git-bug-test` which has a github mirror at
+`cheshirekow.com/git-bug-test`. Ideally we would specify in our git-config
+something like this:
+
+```config
+[git-bug "bridge.github.github-git-bug-test"]
+	user = cheshirekow
+	project = git-bug-test
+	token = xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+	import-filter="assignee=cheshirekow"
+	import-mod="label=+test,label=+github"
+	export-filter="labels=test&github"
+	export-mod="label=-test,label=-github"
+```
+
+with the following semantics:
+
+**import**: Only issues matching `assignee=cheshirekow` will be queried
+from github. Any issue that is created or updated is given the labels `test`
+and `github` in `git-bug`.
+
+**export**: Only bugs which have *both* labels `test` and `github` will be
+considered for export. When we export the bugs, we remove the labels `test`
+and `github`. Or, rather, we do not synchronize those labels with github.
+
+### Open questions:
+
+To what extent can this implementation be generic between different remote
+bug systems? Each remote system will have different sets of metadata that we
+may or may not be able to filter on.
+
+Should we validate that `import-mod` matches `export-filter`? If not then
+bugs can be pulled but not pushed. Maybe that is desirable?
+
+It looks like git-bug already stores github specific metadata (for the github
+bridge). Does that mean we don't need the `import-mod`? I think we do still
+need it because we want to be able to *find* those bugs after import, and we
+would like them to be collated with bugs we've created locally.
+
+The exporter looks like it doesn't update bugs after the initial export. See
```

Comment:
`out <- core.NewExportNothing(b.Id(), "bug already exported")` should not be there. There is https://github.com/MichaelMure/git-bug/issues/231 open, it will be reworked.